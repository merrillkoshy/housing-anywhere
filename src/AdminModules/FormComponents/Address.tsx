import React, { useEffect, useState } from "react";
import { Form } from "react-bootstrap";
import data from "../../components/country-codes.json";

const Address = ({
	houseNumber,
	setHouseNumber,
	street,
	setStreet,
	city,
	setCity,
	state,
	setState,
	postalCode,
	setPostalCode,
	neighborhood,
	setNeighborhood,
	countryCode,
	setCountryCode,
}: {
	houseNumber: string;
	setHouseNumber: React.Dispatch<React.SetStateAction<string>>;
	street: string;
	setStreet: React.Dispatch<React.SetStateAction<string>>;
	city: string;
	setCity: React.Dispatch<React.SetStateAction<string>>;
	state: string;
	setState: React.Dispatch<React.SetStateAction<string>>;
	postalCode: string;
	setPostalCode: React.Dispatch<React.SetStateAction<string>>;
	neighborhood: string;
	setNeighborhood: React.Dispatch<React.SetStateAction<string>>;
	countryCode: string;
	setCountryCode: React.Dispatch<React.SetStateAction<string>>;
}) => {
	const [filteredArray, setFilteredArray] = useState<any>([]);
	const [panel, openPanel] = useState(false);
	useEffect(() => {
		return () => {
			setFilteredArray([]);
			openPanel(false);
		};
	}, []);
	const checkVal = () => {
		setFilteredArray(
			data.filter((dt: any) => {
				if (dt.name?.toLowerCase().includes(countryCode)) {
					return dt;
				}
				return "";
			})
		);
	};
	return (
		<>
			<Form.Group className="mb-3">
				<Form.Label>House Number</Form.Label>
				<Form.Control
					type="text"
					id="housenumber"
					value={houseNumber}
					required
					onChange={(e) => {
						e.preventDefault();
						setHouseNumber(e.currentTarget.value);
					}}
					placeholder="Enter House Number"
				/>
			</Form.Group>

			<Form.Group className="mb-3">
				<Form.Label>Street</Form.Label>
				<Form.Control
					type="text"
					id="street"
					value={street}
					required
					onChange={(e) => {
						e.preventDefault();
						setStreet(e.currentTarget.value);
					}}
					placeholder="Enter Street Name"
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>City</Form.Label>
				<Form.Control
					type="text"
					id="city"
					value={city}
					required
					onChange={(e) => {
						e.preventDefault();
						setCity(e.currentTarget.value);
					}}
					placeholder="Enter City Name"
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>State</Form.Label>
				<Form.Control
					type="text"
					id="state"
					value={state}
					onChange={(e) => {
						e.preventDefault();
						setState(e.currentTarget.value);
					}}
					placeholder="Enter State Name"
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Postal Code</Form.Label>
				<Form.Control
					type="text"
					id="postalCode"
					value={postalCode}
					required
					onChange={(e) => {
						e.preventDefault();
						setPostalCode(e.currentTarget.value);
					}}
					placeholder="Enter Postal Code"
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Neighborhood</Form.Label>
				<Form.Control
					type="text"
					id="neighborhood"
					value={neighborhood}
					onChange={(e) => {
						e.preventDefault();
						setNeighborhood(e.currentTarget.value);
					}}
					placeholder="Enter Neighborhood Name"
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Country Code</Form.Label>
				<Form.Control
					type="text"
					id="countryCode"
					value={countryCode}
					required
					onKeyPress={checkVal}
					onChange={(e) => {
						e.preventDefault();
						if (e.currentTarget.value !== "") openPanel(true);
						setCountryCode(e.currentTarget.value);
					}}
					placeholder="Just type your Country name, we'll fetch your code"
				/>
				<div className={`d-${panel ? "flex" : "none"} autocomplete-items`}>
					{filteredArray.length
						? filteredArray.map((ar: any, i: number) => {
								return (
									<div
										key={ar.code + i}
										onClick={() => {
											setCountryCode(ar.code);
											setFilteredArray([]);
											openPanel(false);
										}}
									>
										{new Set(ar.code)}
									</div>
								);
						  })
						: null}
				</div>
			</Form.Group>
		</>
	);
};
export default Address;
