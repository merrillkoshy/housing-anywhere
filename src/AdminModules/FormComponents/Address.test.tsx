import React, { SetStateAction } from "react";
import { shallow } from "enzyme";
import Address from "./Address";

describe("Address module", () => {
	const AddressModule = shallow(
		<Address
			houseNumber={""}
			setHouseNumber={function(value: SetStateAction<string>): void {
				throw new Error("Function not implemented.");
			}}
			street={""}
			setStreet={function(value: SetStateAction<string>): void {
				throw new Error("Function not implemented.");
			}}
			city={""}
			setCity={function(value: SetStateAction<string>): void {
				throw new Error("Function not implemented.");
			}}
			state={""}
			setState={function(value: SetStateAction<string>): void {
				throw new Error("Function not implemented.");
			}}
			postalCode={""}
			setPostalCode={function(value: SetStateAction<string>): void {
				throw new Error("Function not implemented.");
			}}
			neighborhood={""}
			setNeighborhood={function(value: SetStateAction<string>): void {
				throw new Error("Function not implemented.");
			}}
			countryCode={""}
			setCountryCode={function(value: SetStateAction<string>): void {
				throw new Error("Function not implemented.");
			}}
		/>
	);
	it("renders", () => {
		expect(AddressModule).toMatchSnapshot();
	});

	it("finds a housenumber input which is a string", () => {
		expect(AddressModule.find("#housenumber").exists()).toBe(true);
	});
	it("finds a street input which is a string", () => {
		expect(AddressModule.find("#street").exists()).toBe(true);
	});
	it("finds a city input which is a string", () => {
		expect(AddressModule.find("#city").exists()).toBe(true);
	});
	it("finds a state input which is a string", () => {
		expect(AddressModule.find("#state").exists()).toBe(true);
	});
	it("finds a postalCode input which is a string", () => {
		expect(AddressModule.find("#postalCode").exists()).toBe(true);
	});
	it("finds a neighborhood input which is a string", () => {
		expect(AddressModule.find("#neighborhood").exists()).toBe(true);
	});
	it("finds a countryCode input which is a string", () => {
		expect(AddressModule.find("#countryCode").exists()).toBe(true);
	});
});
