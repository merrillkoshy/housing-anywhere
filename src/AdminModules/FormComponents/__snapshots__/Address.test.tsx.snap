// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address module renders 1`] = `
<Fragment>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      House Number
    </FormLabel>
    <FormControl
      id="housenumber"
      onChange={[Function]}
      placeholder="Enter House Number"
      required={true}
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Street
    </FormLabel>
    <FormControl
      id="street"
      onChange={[Function]}
      placeholder="Enter Street Name"
      required={true}
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      City
    </FormLabel>
    <FormControl
      id="city"
      onChange={[Function]}
      placeholder="Enter City Name"
      required={true}
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      State
    </FormLabel>
    <FormControl
      id="state"
      onChange={[Function]}
      placeholder="Enter State Name"
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Postal Code
    </FormLabel>
    <FormControl
      id="postalCode"
      onChange={[Function]}
      placeholder="Enter Postal Code"
      required={true}
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Neighborhood
    </FormLabel>
    <FormControl
      id="neighborhood"
      onChange={[Function]}
      placeholder="Enter Neighborhood Name"
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Country Code
    </FormLabel>
    <FormControl
      id="countryCode"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Just type your Country name, we'll fetch your code"
      required={true}
      type="text"
      value=""
    />
    <div
      className="d-none autocomplete-items"
    />
  </FormGroup>
</Fragment>
`;
